### ISSUE TITLE
scalapack_05 hangs on ppocon

### ISSUE THREAD

**Issue by davydden** [dealii/dealii#6740]
my build with `netlib-scalapack` + `openblas` + `openmpi` hangs with 4 MPI cores run and matrix `32x32` with block-size `32`, `float` on `ppocon/pspocon_`  https://github.com/dealii/dealii/blob/master/source/lac/scalapack.cc#L1938-L1950 . 

Since the size is so small, we have only 1 active process (1x1 grid) out of 4. And that guy hangs. I will try digging more, but so far it loos like a bug in `netlib-scalapack` as:

* `double` version runs ok for same combination of sizes and block sizes.
* `float` runs ok for combinations
```
  const std::vector<unsigned int> sizes = {{64,120,320,640}};
  const std::vector<unsigned int> blocks = {{32}};
```

**Comment by davydden** [comment]
> Have you gotten a backtrace?

I tried running and noticed a strange backtrace:

```
    frame #7: 0x000000011ff3f381 libmpi.40.dylib`PMPI_Allreduce + 545
    frame #8: 0x000000011fe4427b libmpi_mpifh.40.dylib`mpi_allreduce_ + 123
    frame #9: 0x000000011848444e libparpack.2.dylib`pslamch_ + 142 <- SHALL NOT BE HERE!!!
    frame #10: 0x000000011e21f506 libscalapack.dylib`pspocon_ + 806
    frame #11: 0x000000010371bb69 libdeal_II.g.9.1.0-pre.dylib`dealii::ScaLAPACKMatrix<float>::reciprocal_condition_number(this=0x00007ffeefbfd6c0, a_norm=<unavailable>) const at scalapack.cc:1381
    frame #12: 0x000000010002bc67 scalapack_05.debug`void test<float>(size=<unavailable>, block_size=<unavailable>) at scalapack_05.cc:63
    frame #13: 0x000000010002b93e scalapack_05.debug`main(argc=1, argv=0x00007ffeefbfd8d8) at scalapack_05.cc:81
    frame #14: 0x00007fff6c5ce015 libdyld.dylib`start + 1
    frame #15: 0x00007fff6c5ce015 libdyld.dylib`start + 1
```

It got into running `pArpack`. This reminded me that I actually fixed this other copy-pasted Scalapack function in `Arpack` https://github.com/opencollab/arpack-ng/pull/85  I will install the just released `Arpack@3.6.0` and give it another try...

**Comment by davydden** [comment]
yes, that was indeed the source of the problem
```
$ ctest -R "scalapack_05"
Test project /Users/davydden/libs-sources/deal.ii/deal.ii/_build
    Start 3096: scalapack/scalapack_05.mpirun=1.debug
1/6 Test #3096: scalapack/scalapack_05.mpirun=1.debug ......   Passed    2.16 sec
    Start 3097: scalapack/scalapack_05.mpirun=1.release
2/6 Test #3097: scalapack/scalapack_05.mpirun=1.release ....   Passed    8.36 sec
    Start 3098: scalapack/scalapack_05.mpirun=11.debug
3/6 Test #3098: scalapack/scalapack_05.mpirun=11.debug .....   Passed    6.10 sec
    Start 3099: scalapack/scalapack_05.mpirun=11.release
4/6 Test #3099: scalapack/scalapack_05.mpirun=11.release ...   Passed    5.99 sec
    Start 3100: scalapack/scalapack_05.mpirun=4.debug
5/6 Test #3100: scalapack/scalapack_05.mpirun=4.debug ......   Passed    0.49 sec
    Start 3101: scalapack/scalapack_05.mpirun=4.release
6/6 Test #3101: scalapack/scalapack_05.mpirun=4.release ....   Passed    2.01 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) =  26.16 sec
```
